src/main/java/bio/overture/ego/service/TokenService.java

```diff
@@ -345,23 +351,39 @@ public TokenScopeResponse checkToken(String authToken, String token) {
     return new TokenScopeResponse(owner.getName(), clientId, t.getSecondsUntilExpiry(), names);
   }
 
-  public void revokeToken(UUID userId, @NonNull String tokenName) {
+  public void revokeToken(@NonNull String tokenName) {
     validateTokenName(tokenName);
+    val principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
 
-    log.info(format("Looking for user: '%s'. ", str(userId)));
-    val user =
-        userService
-            .findById(userId)
-            .orElseThrow(
-                () -> new UsernameNotFoundException(format("Can't find user '%s'", str(userId))));
+    if (principal instanceof User) {
+      revokeTokenAsUser(tokenName);
+    } else if (principal instanceof Application) {
+      revokeTokenAsApplication(tokenName, (Application) principal);
+    } else {
+      log.info("Unknown type of authentication, token is not allowed to be revoked.");
+      throw new InvalidRequestException("Unknown type of authentication.");
+    }
+  }
 
-    log.info(format("validating if user '%s' has permission to revoke token.", str(userId)));
-    if (userService.isAdmin(user) && userService.isActiveUser(user)) {
-      revokeToken(tokenName);
+  private void revokeTokenAsUser(String tokenName) {
+    val token =
+        findByTokenString(tokenName)
+            .orElseThrow(() -> new InvalidTokenException("Token not found! "));
+    if (userService.isAdmin(token.getOwner()) && userService.isActiveUser(token.getOwner())) {
```

Comment:
We should be checking if the current principal (requesting user) is an admin, not the token owner.